src/redux/actions.js

```diff
@@ -15,3 +15,27 @@ export const saveActiveItem = (id) => {
     payload: { id: id },
   };
 };
+
+export const fetchServerListWithRouting =
+  (navigate, cookies) => async (dispatch) => {
+    const token = cookies.get('authToken', { path: '/' });
+    if (!token) {
+      navigate('login');
+      console.log('no token');
+    } else {
+      try {
+        const response = await fetch('http://localhost:80/servers', {
+          method: 'GET',
+          headers: {
+            Authorization: `Bearer ${token}`,
+            'Content-Type': 'application/json',
+          },
+        });
+        const data = await response.json();
+        const servers = data.data;
+        dispatch(creationServers(servers));
+      } catch (error) {
+        navigate('login');
```

Comment:
try to check for specific error, because we can achieve another type of error, for example, invalid data or something else, and user will get redirected into the login page